### PR TITLE
DOP-1534: Don't lowercase labels

### PR DESCRIPTION
**Pull request by i80and**
Also finally introduces a more granular postprocessor testing framework, and makes running the parser outside of a git repo acceptable.

### CODE REVIEW

File: snooty/parser.py [hunk 1]
```diff
@@ -1005,9 +1005,17 @@ def __init__(
...
+            branch = subprocess.check_output(
+                ["git", "rev-parse", "--abbrev-ref", "HEAD"],
+                cwd=root,
+                encoding="utf-8",
+                stderr=subprocess.PIPE,
+            ).strip()
+        except subprocess.CalledProcessError as err:
+            logger.info("git error getting branch name: %s", err.stderr)
+            branch = "current"
+
```
Comment (sophstad) [inline snooty/parser.py:1018]: Just a note, would be nice to abstract this from the main parser process eventually. It feels somewhat specific to the MongoDB backend approach we are using.

File: snooty/test_postprocess_granular.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+"""An alternative and more granular approach to writing postprocessing tests.
+   Eventually most postprocessor tests should probably be moved into this format."""
```
Comment (sophstad) [inline snooty/test_postprocess_granular.py:2]: Let's make _this_ file `test_postprocess.py` and call the old one something more curmudgeonly to encourage new work here!

File: snooty/parser.py (hunk 1 above)
Comment (i80and) [reply snooty/parser.py:1018]: oh YES absolutely!

This is a quirk that we should rethink. But right now, I just want it to not be a *fatal* quirk 😸

File: snooty/test_postprocess_granular.py (hunk 2 above)
Comment (i80and) [reply snooty/test_postprocess_granular.py]: Good call!! Updated!

**Review by sophstad — APPROVED** [review]

**Pull request merged**